sklearn/utils/estimator_checks.py

```diff
@@ -509,23 +616,41 @@ def parametrize_with_checks(estimators, *, legacy: bool = True):
         )
         raise TypeError(msg)
 
-    def checks_generator():
+    def _checks_generator(estimators, legacy, expected_failed_checks):
         for estimator in estimators:
-            # First check that the estimator is cloneable which is needed for the rest
-            # of the checks to run
-            name = type(estimator).__name__
-            yield estimator, partial(check_estimator_cloneable, name)
-            for check in _yield_all_checks(estimator, legacy=legacy):
-                check_with_name = partial(check, name)
-                for check_instance in _yield_instances_for_check(check, estimator):
-                    yield _maybe_mark_xfail(check_instance, check_with_name, pytest)
+            args = {"estimator": estimator, "legacy": legacy, "mark": "xfail"}
+            if callable(expected_failed_checks):
+                args["expected_failed_checks"] = expected_failed_checks(estimator)
+            yield from estimator_checks_generator(**args)
 
     return pytest.mark.parametrize(
-        "estimator, check", checks_generator(), ids=_get_check_estimator_ids
+        "estimator, check",
+        _checks_generator(estimators, legacy, expected_failed_checks),
+        ids=_get_check_estimator_ids,
     )
 
 
-def check_estimator(estimator=None, generate_only=False, *, legacy: bool = True):
+@validate_params(
+    {
+        "generate_only": ["boolean"],
+        "legacy": ["boolean"],
+        "expected_failed_checks": [dict, None],
+        "on_skip": [StrOptions({"warn"}), None],
+        "on_fail": [StrOptions({"raise", "warn"}), None],
+        "callback": [callable, None],
+    },
+    prefer_skip_nested_validation=False,
+)
+def check_estimator(
+    estimator=None,
+    generate_only=False,
+    *,
+    legacy: bool = True,
+    expected_failed_checks: dict[str, str] | None = None,
+    on_skip: Literal["warn"] | None = "warn",
+    on_fail: Literal["raise", "warn"] | None = "raise",
+    callback: Callable | None = None,
```

Comment:
Since we have the parameter validation, I'm not sure that the typing is useful.

Reply:
mypy, and IDE / pylance validations, which I find very useful. They don't hurt anyway. I've been adding them on things I add.